From c0f40c20a0d592e873ec4797fcfee0d6c900481c Mon Sep 17 00:00:00 2001
From: Biwen Li <biwen.li@nxp.com>
Date: Fri, 28 Jun 2019 16:36:20 +0800
Subject: [PATCH 551/741] arm: dts: ls1021a: replace ftm0 with ftm_alarm0 DT
 node

commit 265352fd5b557a79c7b46b5618dd70f57a0eefc7 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The patch replaces ftm0 with ftm_alarm0 DT node
	- remove old ftm0 node
	- add rcpm node
	- add ftm_alarm0 node
	- aliases ftm_alarm0 as rtc1

Signed-off-by: Biwen Li <biwen.li@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/boot/dts/ls1021a.dtsi | 26 +++++++++++++++-----------
 1 file changed, 15 insertions(+), 11 deletions(-)

diff --git a/arch/arm/boot/dts/ls1021a.dtsi b/arch/arm/boot/dts/ls1021a.dtsi
index 9d0622fd5b9f..26f3b66b61f0 100644
--- a/arch/arm/boot/dts/ls1021a.dtsi
+++ b/arch/arm/boot/dts/ls1021a.dtsi
@@ -25,6 +25,7 @@
 		serial4 = &lpuart4;
 		serial5 = &lpuart5;
 		sysclk = &sysclk;
+		rtc1 = &ftm_alarm0;
 	};
 
 	cpus {
@@ -625,17 +626,6 @@
 			status = "disabled";
 		};
 
-		ftm0: ftm0@29d0000 {
-			compatible = "fsl,ls1021a-ftm-alarm";
-			reg = <0x0 0x29d0000 0x0 0x10000>,
-			      <0x0 0x1ee2144 0x0 0x4>,
-			      <0x0 0x0157051c 0x0 0x4>;
-			reg-names = "ftm", "pmctrl", "scrachpad";
-			interrupts = <GIC_SPI 118 IRQ_TYPE_LEVEL_HIGH>;
-			big-endian;
-			status = "okay";
-		};
-
 		wdog0: watchdog@2ad0000 {
 			compatible = "fsl,imx21-wdt";
 			reg = <0x0 0x2ad0000 0x0 0x10000>;
@@ -984,5 +974,19 @@
 			big-endian;
 		};
 
+                rcpm: rcpm@1ee2140 {
+                        compatible = "fsl,ls1021a-rcpm", "fsl,qoriq-rcpm-2.1+";
+                        reg = <0x0 0x1ee2140 0x0 0x8>;
+                        #fsl,rcpm-wakeup-cells = <2>;
+                };
+
+                ftm_alarm0: timer0@29d0000 {
+                        compatible = "fsl,ls1021a-ftm-alarm";
+                        reg = <0x0 0x29d0000 0x0 0x10000>;
+                        fsl,rcpm-wakeup = <&rcpm 0x0 0x20000000>;
+                        interrupts = <GIC_SPI 118 IRQ_TYPE_LEVEL_HIGH>;
+                        big-endian;
+                };
+
 	};
 };
-- 
2.17.1

